environment/src/main/java/com/sequenceiq/environment/environment/flow/deletion/handler/ResourceEncryptionDeleteHandler.java

```diff
@@ -60,12 +58,7 @@ public void accept(Event<EnvironmentDeletionDto> environmentDtoEvent) {
         try {
             environmentService.findEnvironmentById(environmentDto.getId()).ifPresent(environment -> {
                 if (AZURE.name().equalsIgnoreCase(environmentDto.getCloudPlatform())) {
-                    String diskEncryptionSetId = Optional.ofNullable(environmentDto.getParameters())
-                            .map(ParametersDto::getAzureParametersDto)
-                            .map(AzureParametersDto::getAzureResourceEncryptionParametersDto)
-                            .map(AzureResourceEncryptionParametersDto::getDiskEncryptionSetId)
-                            .orElse(null);
-                    if (StringUtils.isNotEmpty(diskEncryptionSetId) &&
+                    if (entitlementService.isAzureDiskSSEWithCMKEnabled(environmentDto.getAccountId()) &&
```

Comment:
As noted in the Jira ticket, I think the right approach is to check for the presence of `AzureResourceEncryptionParametersDto::getEncryptionKeyUrl` instead of `AzureResourceEncryptionParametersDto::getDiskEncryptionSetId`. That logic used to be present since #10740, but it got later reverted in #11225 for an unknown reason.

While the entitlement check may seem as a quick-and-dirty fix, it is quite inefficient. With this, deletion will be always attempted if the entitlement is granted, even if the DTO does not specify any encryption key URL at all.